src/home_robot/home_robot/utils/bullet.py

```diff
@@ -460,20 +466,26 @@ def compute_fk(self, q: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
 
     def compute_ik(
         self,
-        pos_desired,
-        quat_desired,
-        q_init: np.ndarray = None,
-        num_attempts: int = 5,
+        pos_desired: np.ndarray,
+        quat_desired: np.ndarray,
+        q_init: Optional[np.ndarray] = None,
+        num_attempts: int = 1,
         verbose: bool = False,
-    ):
+        **kwargs,
+    ) -> Tuple[np.ndarray, bool, dict]:
+        q_out = None
+        success = False
+
         if q_init is not None:
             # This version assumes that q_init is NOT in the right format yet
             self.set_joint_positions(q_init)
             num_attempts = 1
         elif self.controlled_joints is not None and self.range is not None:
+            # TODO: this q_init isn't used? Should this logic be before self.set_joint_positions?
             rng = self.range[:, 1] - self.range[:, 0]
             rng[np.isinf(rng)] = 0
             q_init = (np.random.random() * rng) + self.range[:, 0]
```

Comment:
This q_init isn't used? Should this logic be before self.set_joint_positions?

Reply:
discussing offline